packages/application-shell/src/components/locale-switcher/locale-switcher.tsx

```diff
@@ -79,22 +105,45 @@ const LocaleSwitcher = (props: Props) => {
         name="locale-switcher"
         aria-labelledby={LOCALE_SWITCHER_LABEL_ID}
         onChange={handleSelection}
-        options={props.availableLocales.map((locale) => ({
-          label: locale,
-          value: locale,
-        }))}
+        options={localeOptions}
         components={{
           SingleValue,
           ValueContainer: PatchedValueContainer,
           MenuList: CustomMenuList,
+          GroupHeading: (groupProps) => (
+            <CustomGroupHeading
+              {...groupProps}
+              setIsOpen={setIsDescriptionDialogOpened}
+            />
+          ),
         }}
         isClearable={false}
         backspaceRemovesValue={false}
         isSearchable={false}
         horizontalConstraint={'auto'}
         appearance="quiet"
         maxMenuHeight={360}
+        minMenuWidth={3}
       />
+      {/* Dialog that explains the locales */}
+      <InfoDialog
+        isOpen={isDescriptionDialogOpened}
+        title="Selecting a data locale"
+        onClose={() => setIsDescriptionDialogOpened(false)}
+      >
+        <Spacings.Stack scale="xl">
+          <div>
+            The selected data locale will be shown as default in all localized
+            fields in the Merchant Center, like names, descriptions, or
+            localized attributes.
+          </div>
+          <div>
+            Note that this does not affect the interface language of the
+            Merchant Center and the data formatting options. To change this, go
+            to user profile.
+          </div>
```

Comment:
⚠️ This should go into an i18n message. It should also be defined as one message, not two.

Reply:
But the thing is we want to have a spacing in the dialog between the two sentences. 🤔 
![Screenshot 2024-02-20 at 10 41 58](https://github.com/commercetools/merchant-center-application-kit/assets/52276952/c8ae86c8-af5a-426f-a8ec-a04f8751d74f)

Reply:
For that we can use the [rich-text formatting](https://formatjs.io/docs/core-concepts/icu-syntax#rich-text-formatting).

In the message you can use something like `<newline></newline>`, then pass it as a `values`

```
First sentence.<newline></newline><newline></newline>Second sentence.
```

Then the values

```ts
values: getNewLine
```

And the function

```ts
const getNewLine = () => <br />;
```

You can see in our internal applications some examples of these

Reply:
Got it! Thanks for explanation. [72203be](https://github.com/commercetools/merchant-center-application-kit/pull/3403/commits/72203be0940559403ea508b66d47026e7202938f)